### ISSUE TITLE
New MT6 install: provision roundcube fails

### ISSUE THREAD

**Issue by johannes73** [msimerson/Mail-Toaster-6#194]
### Expected behavior

provision roundcube
[...]
"Success..."  

### Observed behavior

provision roundcube
[...]

   ***   starting nginx   ***

sysrc -R /jails/stage nginx_enable=YES
nginx_enable:  -> YES
jexec stage service nginx start
Performing sanity check on nginx configuration:
nginx: [emerg] open() "/data/etc/nginx.conf" failed (2: No such file or directory)
nginx: configuration file /data/etc/nginx.conf test failed
Starting nginx.
nginx: [emerg] open() "/data/etc/nginx.conf" failed (2: No such file or directory)
/usr/local/etc/rc.d/nginx: WARNING: failed to start nginx
nginx does not exist in /etc/rc.d or the local startup
directories (/usr/local/etc/rc.d)
curl: (7) Failed to connect to 172.16.15.254 port 80: Connection refused

### Steps to reproduce

I continued the fresh install from Issue #193 (I know maybe I shouldn't have, as there was an issue).

To sum it up:

1. Install FreeBSD-10.3-RELEASE to server.
2. Follow all steps on https://github.com/msimerson/Mail-Toaster-6/wiki/FreeBSD, except do not install "install CA $igned certificate", because we will use "provision letsencrypt".
3. Provision all jails, from "dns" to "roundcube", including "letsencrypt".

**Comment by msimerson** [comment]
Did the other webmail jails install?  This is a suspiciously similar (didn't mount /data) issue to your previous one.

**Comment by johannes73** [comment]
Yes, I agree, it is a bit suspicious, as I know I got the webmail apps up and running in an earlier test install (fall 2016). This is a fresh install, Feb 10, so not an upgrade.

These are the results of the provisions of the other webmail apps. In short: roundcube, rainloop, squirrelmail fails, but sqwebmail succeeds.

## provision rainloop

   ***   starting nginx   ***

sysrc -R /jails/stage nginx_enable=YES
nginx_enable:  -> YES
jexec stage service nginx start
Performing sanity check on nginx configuration:
nginx: [emerg] open() "/data/etc/nginx.conf" failed (2: No such file or directory)
nginx: configuration file /data/etc/nginx.conf test failed
Starting nginx.
nginx: [emerg] open() "/data/etc/nginx.conf" failed (2: No such file or directory)
/usr/local/etc/rc.d/nginx: WARNING: failed to start nginx
nginx does not exist in /etc/rc.d or the local startup
directories (/usr/local/etc/rc.d)

   ***   testing nginx is running   ***

checking for process nginx in staged jail

[ ...exits... ]

## provision squirrelmail

   ***   starting nginx   ***

sysrc -R /jails/stage nginx_enable=YES
nginx_enable:  -> YES
jexec stage service nginx start
Performing sanity check on nginx configuration:
nginx: [emerg] open() "/data/etc/nginx.conf" failed (2: No such file or directory)
nginx: configuration file /data/etc/nginx.conf test failed
Starting nginx.
nginx: [emerg] open() "/data/etc/nginx.conf" failed (2: No such file or directory)
/usr/local/etc/rc.d/nginx: WARNING: failed to start nginx
nginx does not exist in /etc/rc.d or the local startup
directories (/usr/local/etc/rc.d)

   ***   testing nginx is running   ***

checking for process nginx in staged jail

[ ... exits... ]

## provision sqwebmail

   ***   service jail start sqwebmail   ***

Starting jails: sqwebmail.

Success! A new 'sqwebmail' jail is provisioned

**Comment by msimerson** [comment]
No surprise, same issues. I've been testing only with 11.0. I'll spin up a 10.3 instance and see if I can't replicate your issue.

**Comment by johannes73** [comment]
Great, I would go for 11.0, but do not dare yet, because I received some nasty core dumps on my server  (AMD) when I tested 11.0... So I'm sticking with 10.3 for now...

**Comment by msimerson** [comment]
I'm not able to replicate this one:

````

   ***   adding roundcube to /etc/jail.conf   ***


roundcube	{
		ip4.addr = lo1|172.16.15.33;
		mount += "/data/roundcube $path/data nullfs rw 0 0";
	}

   ***   service jail start roundcube   ***

Starting jails: roundcube.

Success! A new 'roundcube' jail is provisioned

[root@freebsd10 ~/mt6]# uname -a
FreeBSD freebsd10.imac27.simerson.net 10.3-RELEASE-p11 FreeBSD 10.3-RELEASE-p11 #0: Mon Oct 24 18:49:24 UTC 2016     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GENERIC  amd64
[root@freebsd10 ~/mt6]# 
````

**Comment by johannes73** [comment]
Thanks for checking. Now it is working after a hack in my nginx.conf files... It seems the "haproxy" did not get resolved to IP address(?). It works after I replaced "haproxy" with the hardcoded IP.

```
# cd /data/roundcube/etc
# ls 
nginx-locations.conf	nginx.conf		nginx.conf.orig
# diff nginx.conf.orig nginx.conf
18c18
< 	set_real_ip_from haproxy;
---
> 	set_real_ip_from 172.16.15.12;
```

And the same for:
/data/rainloop/etc/nginx.conf
/data/squirrelmail/etc/nginx.conf

Not sure why "haproxy" did not get resolved... 

The dns jail is running:
```
# jls | grep dns
    60  172.16.15.3     dns                           /jails/dns
```

Running as root "haproxy" gets resolved from the dns jail:

```
# drill haproxy
;; ->>HEADER<<- opcode: QUERY, rcode: NOERROR, id: 44183
;; flags: qr aa rd ra ; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0 
;; QUESTION SECTION:
;; haproxy.	IN	A

;; ANSWER SECTION:
haproxy.	3600	IN	A	172.16.15.12

;; AUTHORITY SECTION:

;; ADDITIONAL SECTION:

;; Query time: 0 msec
;; SERVER: 172.16.15.3
;; WHEN: Sat Feb 11 04:11:20 2017
;; MSG SIZE  rcvd: 41

```

**Comment by johannes73** [comment]
After these adjustments to the nginx.conf files, it works. I am not sure whether I made something wrong during the install, that caused this, or if this could be fixed somehow in the install scripts.

Matt, a big thank you for your *super* support today. I really appreciate it. Now the MT6 is up and running, including rainloop and squirrelmail. roundcube is not running, it says:

"DATABASE ERROR: CONNECTION FAILED!"

**Comment by Infern1** [comment]
Just as background, using 10.3 also, not had these kind of problems.

**Comment by msimerson** [comment]
Closing, as there's no way to advance this further. If new information comes to light, or if you determine what the issue(s) were, please do post them here.
